ams/server/src/main/java/com/netease/arctic/server/optimizing/maintainer/IcebergTableMaintainer.java

```diff
@@ -100,9 +100,13 @@ public void cleanOrphanFiles(TableRuntime tableRuntime) {
 
     // refresh
     table.refresh();
-
-    // clear dangling delete files
-    cleanDanglingDeleteFiles();
+    String version = table.properties().get("format-version");
```

Comment:
The `format-version` is not a requied properties, so it may bring NPE.

Additional, the defalut value of `format-version` is `1` before iceberg version 1.4.0, and changed to `2` after that ( related [PR](https://github.com/apache/iceberg/pull/8381)), so it may not be a stable way.

Maybe we can use the summary(the count of delete files) of the current snapsnot to determine whether we should cleaning the dangling delete files, and if the current snapshot is null, we also shouldn't cleaning them.

Reply:
`org.apache.iceberg.SnapshotSummary#TOTAL_DELETE_FILES_PROP` is the key of snapshot summary which represents the count of delete filesl

Reply:
Then do we need to make additional judgments on the format?